tasks/shell_crmsh/create-and-push-cib.yml

```diff
@@ -0,0 +1,266 @@
+# SPDX-License-Identifier: MIT
+---
+# We always need to create CIB to see whether it's the same as what is already
+# present in the cluster. However, we don't want to report it as a change since
+# the only thing which matters is pushing the resulting CIB to the cluster.
+
+# Create backup of current cib in /root
+- name: Gather facts for ansible_date_time
+  ansible.builtin.setup:
+    filter:
+      - 'ansible_date_time'
+
+# Prepare CIB files
+- name: Create a tempfile for original CIB
+  ansible.builtin.tempfile:
+    state: file
+    suffix: _ha_cluster_original_cib_xml
+  register: __ha_cluster_tempfile_original_cib_xml
+  check_mode: false
+  changed_when: not ansible_check_mode
+
+- name: Create a tempfile for new CIB
+  ansible.builtin.tempfile:
+    state: file
+    suffix: _ha_cluster_cib_xml
+  register: __ha_cluster_tempfile_cib_xml
+  check_mode: false
+  changed_when: not ansible_check_mode
+
+# Maintenance mode is required, because CIB version changes with cluster
+# status changes, resulting in shadow CIB outdated and unable to patch.
+# Sleep is implemented to ensure that cluster have enough time to freeze
+# to ensure CIB export consistency.
+- name: Put cluster in maintenance mode to freeze cib changes
```

Comment:
@tomjelinek This also seems like good logic to follow; even if the Ansible Role was executed on a Linux Pacemaker Cluster running atop hosts for 'PROD' of something, putting into maintenance mode is a industry-wide best practice for any in-place software re-configuration/update?

Is this used elsewhere? As an end-user, I would expect commonality of approach between the different PCMK Shells - although it's not a deal breaker as there will always be some variance between the PCMK Shells.

Reply:
Sorry, I don't follow. Why would it be needed to put the cluster into the maintenance mode?

When using crmsh manually from command line, is the user required to put a cluster into the maintenance mode before doing any changes to the cluster configuration? I haven't heard that pacemaker would require that. Corosync doesn't have a maintenance mode at all, how would you change its configuration then?

Reply:
cibadmin --query is sensitive on Time and cluster status, which means that I was able to reproduce 20-50% chance for cibadmin patch to result in error with outdated version, just because cib build workflow was working on already outdated cib export.

This step freezes cluster, waits for it to reflect freeze and then gets accurate cib, which will be accepted in the end.

Reply:
This should be prevented by specifying `--no-version` option for `crm_diff`, which both pcs and crmsh parts do. I haven't run into the outdated cib error even once with the role and it wasn't reported by anyone, as far as I know, either.

Reply:
> Sorry, I don't follow. Why would it be needed to put the cluster into the maintenance mode?

@tomjelinek My original comment was focused on the general principle/usage of any "maintenance mode" as an industry-wide best practice for any in-place software re-configuration/update. Seems logical if the maintenance mode exists, we should use it as intended. Clearly the Ansible Role has worked (very well) to date, without needing this. I was only highlighting whether it would be good to include in `pcs` PCMK Shell execution flow, as it seems logical.

.... I guess the question becomes, `"Why do we NOT use 'pcs' PCMK Shell in maintenance mode before making changes?"`

---

Although from the continuing discussion with @marcelmamula , it looks like maintenance mode is a firm requirement when using `crmsh` PCMK Shell.

Reply:
Maintenance mode was enhanced to cover edge case with is-managed meta attribute

Reply:
There is no action remaining here, only @tomjelinek to respond to posterity

Reply:
As I mentioned, there haven't been any issues caused by the role not utilizing maintenance mode with pcs, to my knowledge. If maintenance mode is required for the role to run correctly with crmsh, then use it, of course. I'll take a look at the scenario @marcelmamula described to see if I can reproduce the issue using pcs and whether using maintenance mode fixes that.